### PR TITLE
fix: loosen peerDependencies requirement on mongodb

### PR DESCRIPTION
**Pull request by addaleax**
See [NODE-3108](https://jira.mongodb.org/browse/NODE-3108).

### CODE REVIEW

**Review by nbbeeken — APPROVED** [review]
LGTM, I wonder if we want to make this `>=3.6` to suggest a supported driver version.
Maybe that's not the place for that though, I'll let the team chime in.
Does npm 7 now install peerDeps by default?

**Comment by addaleax** [comment]
> Does npm 7 now install peerDeps by default?

Yes, that’s what happens here :+1:

**Review by durran — APPROVED** [review]

**Review by emadum — APPROVED** [review]

**Pull request merged**
